File: src/mpmt_mss/client/python/fpga_example.py
from mssclient import MSSClient

client = MSSClient("http://zynq:8000/rpc")

reg = 1
value = client.fpga.readRegister(reg)
print(f'reg({reg}), {value}, {hex(value)}')

reg = 0
value = client.fpga.readRegister(reg)
print(f'reg({reg}), {value}, {hex(value)}')

value = client.fpga.writeRegister(0, 5)

value = client.fpga.readRegister(reg)
print(f'reg({reg}), {value}, {hex(value)}')

File: src/mpmt_mss/client/cpp/examples/fpga_example.cpp
#include <iostream>

#include <mpmt_mss/mssclient.hpp>

// C++ port of client/python/fpga_example.py

using namespace mpmt_mss;

int main() {
  MSSClient client("http://zynq:8000/rpc");

  int64_t reg = 1;
  int64_t value = client.fpga.readRegister(reg);
  std::cout << "reg(" << reg << "), " << value << ", 0x" << std::hex << value << std::dec
            << std::endl;

  reg = 0;
  value = client.fpga.readRegister(reg);
  std::cout << "reg(" << reg << "), " << value << ", 0x" << std::hex << value << std::dec
            << std::endl;

  client.fpga.writeRegister(0, 5);

  value = client.fpga.readRegister(reg);
  std::cout << "reg(" << reg << "), " << value << ", 0x" << std::hex << value << std::dec
            << std::endl;

  return 0;
}
